src/dawn/engine_dawn.cc

```diff
@@ -659,7 +688,11 @@ Result DawnPipelineHelper::CreateRenderPipelineDescriptor(
     depth_stencil_format = ::dawn::TextureFormat::D32FloatS8Uint;
   }
 
-  renderPipelineDescriptor.layout = MakeBasicPipelineLayout(device, nullptr);
+  if (render_pipeline.hasBinding)
```

Comment:
I think you can replace this "if" by passing down a reference to render_pipeline.bind_group_layout down to the  helper, and then the helper can do different things based on whether the layout compares equal to false or not.

Reply:
ok, we can leave this for a future cleanup,.